### PR TITLE
emoji

### PR DESCRIPTION
**Pull request by LetterN**
adds emojis.

### CODE REVIEW

**Comment by JJRcop** [comment]
🙂

File: code/modules/admin/verbs/adminpm.dm [hunk 1]
```diff
@@ -36,41 +36,46 @@
...
-			if(holder)	to_chat(src, "<font color='red'>Error: Admin-PM: Client not found.</font>")
-			else		to_chat(src, "<font color='red'>Error: Private-Message: Client not found. They may have lost connection, so try using an adminhelp!</font>")
+			if(holder)
+				to_chat(src, "<font color='red'>Error: Admin-PM: Client not found.</font>")
+			else
+				to_chat(src, "<font color='red'>Error: Private-Message: Client not found. They may have lost connection, so try using an adminhelp!</font>")
 			return
 
-	if (src.handle_spam_prevention(msg,MUTE_ADMINHELP))
+	if(src.handle_spam_prevention(msg,MUTE_ADMINHELP))
```
Comment (ninjanomnom) [inline code/modules/admin/verbs/adminpm.dm]: remove the `src.` here while you're at it

File: code/modules/admin/verbs/adminpm.dm [hunk 2]
```diff
@@ -36,41 +36,46 @@
...
 		return
 
 	//clean the message if it's not sent by a high-rank admin
 	if(!check_rights(R_SERVER|R_DEBUG,0))
-		msg = sanitize(copytext(msg,1,MAX_MESSAGE_LEN))
-		if(!msg)	return
+		msg = trim(sanitize(copytext(msg,1,MAX_MESSAGE_LEN)))
+		if(!msg)
+			return
+		msg = emoji_parse(msg)
```
Comment (ninjanomnom) [inline code/modules/admin/verbs/adminpm.dm]: High rank admins don't get emojis in pms?

**Comment by JJRcop** [comment]
Well, that's sad

**Pull request closed without merging**